controllers/buildpipeline/buildpipeline_adapter.go

```diff
@@ -132,7 +132,8 @@ func (a *Adapter) EnsureSnapshotExists() (controller.OperationResult, error) {
 	err = a.client.Create(a.context, expectedSnapshot)
 	if err != nil {
 		a.logger.Error(err, "Failed to create Snapshot")
-		return controller.RequeueWithError(err)
+		// we cannot create a snapshot and we don't want to block resources, user has to retry
+		return a.removeFinalizerAndStopProcessing()
```

Comment:
We need to check if the `err` is about the reached snapshot quota then remove the finalizer, as I wanted to do in https://github.com/redhat-appstudio/integration-service/pull/481#discussion_r1429759860, I think.
The retry for other err is still needed here.

Reply:
updated